src/Workleap.Extensions.Mongo.Analyzers/IndexAttributeUsageAnalyzer.cs

```diff
@@ -32,18 +32,33 @@ public override void Initialize(AnalysisContext context)
     private static void OnCompilationStarted(CompilationStartAnalysisContext context)
     {
         var analyzer = new IndexAttributeUsageAnalyzerImplementation(context.Compilation);
-        if (analyzer.IsValid)
+        if (analyzer.ShouldScan)
         {
             context.RegisterOperationAction(analyzer.AnalyzeOperationInvocation, OperationKind.Invocation);
         }
     }
 
     private sealed class IndexAttributeUsageAnalyzerImplementation
     {
+        /// <summary>
+        /// Symbol representing the type MongoDB.Driver.IMongoCollectionExtensions, if we see this symbol trigger the validation
```

Comment:
Are those comments really necessary? I feel like reading the code and understanding what's happening should be telling of what it is used for.